recipes/fixed_math/all/conanfile.py

```diff
@@ -0,0 +1,140 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/arturbac/fixed_math/"
+    topics = ("mathematics", "fixed-point")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "header_only": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "header_only": False,
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "10", # fixed_math requires __has_builtin
+            "clang": "7",
+            "apple-clang": "12",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.header_only:
+            self.options.rm_safe("shared")
+            self.options.rm_safe("fPIC")
+            self.package_type = "header-library"
+        elif self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        if self.options.header_only:
+            basic_layout(self, src_folder="src")
+        else:
+            cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        if self.info.options.header_only:
+            self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def build_requirements(self):
+        if not self.options.header_only:
+            self.tool_requires("cmake/[>=3.21 <4]")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        if self.options.header_only:
+            return
+        tc = CMakeToolchain(self)
+        if is_msvc(self):
+            tc.variables["CMAKE_CXX_FLAGS"] = "/Zc:__cplusplus"
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        tc.generate()
+        venv = VirtualBuildEnv(self)
+        venv.generate(scope="build")
+
+    def build(self):
+        replace_in_file(self, os.path.join(self.source_folder, "fixed_lib", "include", "fixedmath", "fixed_math.hpp"),
+            "template<typename arithmethic_type>\n  constexpr fixed_t arithmetic_to_fixed( arithmethic_type value ) noexcept;",
+            "template<typename arithmethic_type, typename>\n  constexpr fixed_t arithmetic_to_fixed( arithmethic_type value ) noexcept;"
```

Comment:
Any insight into this?

Reply:
@AbrilRBS 
Thank you for your comment.
I add comment for this patch.

Reply:
This patch has been merged into upstream.